contracts/DepegMessageHelper.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.2;
+
+import "./EIP712.sol";
+
+
+contract DepegMessageHelper is 
+    EIP712
+{
+    // EIP-712 Depeg specifics
+    string public constant EIP712_DOMAIN_NAME = "EtheriscDepeg";
+    string public constant EIP712_DOMAIN_VERSION = "1";
+
+    string public constant EIP712_POLICY_TYPE = "Policy(address wallet,uint256 protectedBalance,uint256 duration,uint256 bundleId,bytes32 signatureId)";
+    bytes32 private constant EIP712_POLICY_TYPE_HASH = keccak256(abi.encodePacked(EIP712_POLICY_TYPE));
+
+    // tracking of signatures
+    mapping(bytes32 /* signature hash */ => bool /* used */) private _signatureIsUsed;
+
+
+    constructor()
+        EIP712(EIP712_DOMAIN_NAME, EIP712_DOMAIN_VERSION)
+    { }
+
+    function processSignature(
```

Comment:
I don't like the name of the method as it does not indicate that the signature is then also registered as used. maybe `checkAndRegisterSignature` would be better.

Reply:
done